nearcore/src/config.rs

```diff
@@ -723,8 +724,10 @@ impl NearConfig {
             validator_signer.write_to_file(&dir.join(&self.config.validator_key_file));
         }
 
-        let network_signer =
-            InMemorySigner::from_secret_key("".to_string(), self.network_config.secret_key.clone());
+        let network_signer = InMemorySigner::from_secret_key(
+            "node".parse().unwrap(),
+            self.network_config.secret_key.clone(),
+        );
         network_signer.write_to_file(&dir.join(&self.config.node_key_file));
```

Comment:
> I assume there is some part of code which generates this node key which we need to adapt.

I believe that's here, and it's been substituted for `"node"`.

Reply:
It seems like it is throwing this error:
```


[2021-06-30T16:05:02Z] Jun 30 16:04:51.611  INFO neard: Version: 1.2.0, Build: c9752738, Latest Protocol: 45
--
  | [2021-06-30T16:05:02Z] thread 'main' panicked at 'Failed to deserialize KeyFile: Error("the value is too short for account ID", line: 2, column: 18)', core/crypto/src/key_file.rs:36:40
  | [2021-06-30T16:05:02Z] stack backtrace:
  | [2021-06-30T16:05:02Z]    0: rust_begin_unwind
  | [2021-06-30T16:05:02Z]              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:493:5
  | [2021-06-30T16:05:02Z]    1: core::panicking::panic_fmt
  | [2021-06-30T16:05:02Z]              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/panicking.rs:92:14
  | [2021-06-30T16:05:02Z]    2: core::option::expect_none_failed
  | [2021-06-30T16:05:02Z]              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/option.rs:1300:5
  | [2021-06-30T16:05:02Z]    3: core::result::Result<T,E>::expect
  | [2021-06-30T16:05:02Z]              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/result.rs:997:23
  | [2021-06-30T16:05:02Z]    4: near_crypto::key_file::KeyFile::from_file
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/core/crypto/src/key_file.rs:36:9
  | [2021-06-30T16:05:02Z]    5: near_crypto::signer::InMemorySigner::from_file
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/core/crypto/src/signer.rs:62:9
  | [2021-06-30T16:05:02Z]    6: nearcore::config::load_config_without_genesis_records
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/nearcore/src/config.rs:1107:26
  | [2021-06-30T16:05:02Z]    7: near::cli::RunCmd::run
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/neard/src/../src/cli.rs:197:31
  | [2021-06-30T16:05:02Z]    8: near::cli::NeardCmd::parse_and_run
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/neard/src/../src/cli.rs:47:42
  | [2021-06-30T16:05:02Z]    9: near::main
  | [2021-06-30T16:05:02Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0a49689549841a3a9-1/nearprotocol/nearcore/neard/src/../src/main.rs:42:5
  | [2021-06-30T16:05:02Z]   10: core::ops::function::FnOnce::call_once
  | [2021-06-30T16:05:02Z]              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/ops/function.rs:227:5
  | [2021-06-30T16:05:02Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
  | [2021-06-30T16:05:02Z]
  | [2021-06-30T16:05:05Z] node 3 started
  | [2021-06-30T16:05:18Z] Traceback (most recent call last):
  | [2021-06-30T16:05:18Z]   File "tests/sanity/upgradable.py", line 167, in <module>
  | [2021-06-30T16:05:18Z]     main()
  | [2021-06-30T16:05:18Z]   File "tests/sanity/upgradable.py", line 83, in main
  | [2021-06-30T16:05:18Z]     assert 'error' not in res, res
```

so I believe the "node" substitution is not working. I assume it expects most likely some real ed25519 key, but than again confusing why does empty work. Let me check a bit more.

Reply:
Also one other test is failing on this one:
```


[2021-06-30T16:10:14Z] Snapshot: validators_update_transaction
--
  | [2021-06-30T16:10:14Z] Source: chain/rosetta-rpc/src/adapters/mod.rs:992
  | [2021-06-30T16:10:14Z] -old snapshot
  | [2021-06-30T16:10:14Z] +new results
  | [2021-06-30T16:10:14Z] ────────────────────────────────────────────────────────────────────────────────
  | [2021-06-30T16:10:14Z] validators_update_transaction
  | [2021-06-30T16:10:14Z] ────────────┬───────────────────────────────────────────────────────────────────
  | [2021-06-30T16:10:14Z]    12    12 │             status: Some(
  | [2021-06-30T16:10:14Z]    13    13 │                 Success,
  | [2021-06-30T16:10:14Z]    14    14 │             ),
  | [2021-06-30T16:10:14Z]    15    15 │             account: AccountIdentifier {
  | [2021-06-30T16:10:14Z]    16       │-                address: "nfvalidator1.near",
  | [2021-06-30T16:10:14Z]          16 │+                address: AccountId(
  | [2021-06-30T16:10:14Z]          17 │+                    "nfvalidator1.near",
  | [2021-06-30T16:10:14Z]          18 │+                ),
  | [2021-06-30T16:10:14Z]    17    19 │                 sub_account: None,
  | [2021-06-30T16:10:14Z]    18    20 │             },
  | [2021-06-30T16:10:14Z]    19    21 │             amount: Some(
  | [2021-06-30T16:10:14Z]    20    22 │                 Amount {
  | [2021-06-30T16:10:14Z] ┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
  | [2021-06-30T16:10:14Z]    37    39 │             status: Some(
  | [2021-06-30T16:10:14Z]    38    40 │                 Success,
  | [2021-06-30T16:10:14Z]    39    41 │             ),
  | [2021-06-30T16:10:14Z]    40    42 │             account: AccountIdentifier {
  | [2021-06-30T16:10:14Z]    41       │-                address: "nfvalidator2.near",
  | [2021-06-30T16:10:14Z]          43 │+                address: AccountId(
  | [2021-06-30T16:10:14Z]          44 │+                    "nfvalidator2.near",
  | [2021-06-30T16:10:14Z]          45 │+                ),
  | [2021-06-30T16:10:14Z]    42    46 │                 sub_account: None,
  | [2021-06-30T16:10:14Z]    43    47 │             },
  | [2021-06-30T16:10:14Z]    44    48 │             amount: Some(
  | [2021-06-30T16:10:14Z]    45    49 │                 Amount {
  | [2021-06-30T16:10:14Z] ────────────┴───────────────────────────────────────────────────────────────────
  | [2021-06-30T16:10:14Z] thread 'adapters::tests::test_convert_block_changes_to_transactions' panicked at 'snapshot assertion for 'validators_update_transaction' failed in line 992', /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/insta-1.6.3/src/runtime.rs:1026:9
```

I'm not really sure what the issue is. It seems like there are two seperate issues ones seems to be that it is unable to create a `InMemorySigner::from_file` due to something, the other seems like the accountId format is clashing with the old string only version.

I believe best people to talk to are either @bowenwang1996 @abacabadabacaba or @mfornet maybe, as frol is out of the office?

Reply:
Also @EgorKulikov would be a good person as he was working a lot on the AccountId refactor, so could be familiar with this code parts.

Reply:
It seems to me that this tests actually needs to be changed/fixed here: https://github.com/near/nearcore/blob/master/chain/rosetta-rpc/src/adapters/mod.rs#L968. As now the account type is different, but will leave it to them, as they have better understanding.

Reply:
The second issue as pointed out here is caused by having a non-String Debug representation of `AccountId`. It's properly explained in the dropdown here https://github.com/near/nearcore/pull/4440#issuecomment-871606219. We could maintain the previous `"id"` representation or use the new `AccountId("id")` format.

Reply:
In that case I assume it is fine just to implement that. Still than the questions is why does the other failure happen.